activerecord/test/support/method_stub.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module MethodStub
+  # FIXME: Are all these tests for method calls (and not called)
+  # really necessary?
+  private
+
+    def with_stubbed_method(object, method_name, return_value = nil)
+      stub_method(object, method_name, return_value)
+
+      yield
+
+    ensure
+      unstub_method(object, method_name)
+    end
+
+    # TODO: `alias_method` is private in Ruby 2.4, so once the minimum
+    # version is 2.5 there'll be no need to resort to `module_eval`.
+
+    def stub_method(object, method_name, return_value = nil)
```

Comment:
It seems works only if `object` is a Class. If pass an instance as `object` it doesn't work, does it?
I still think that it should be renamed to `with_stubbed_instance_method` since it only can stub method for any instance of the specific class.
For stubbing any method of an object we have `assert_called_with`
@utilum What do you think about it?

Reply:
Why not?